src/main/java/org/openrewrite/java/testing/junit5/MigrateFromHamcrest.java

```diff
@@ -0,0 +1,119 @@
+                      .javaParser(JavaParser.fromJavaVersion().classpathFromResources(executionContext, "junit-jupiter-api-5.9"))
+                      .staticImports("org.junit.jupiter.api.Assertions." + targetAssertion)
+                      .build();
+                    mi = withTemplate(mi, template, mi.getArguments().get(0), stripMatcherInvocation(mi.getArguments().get(1)));
+                    maybeAddImport("org.junit.jupiter.api.Assertions", targetAssertion);
+                }
+                else throw new IllegalArgumentException("Parameter mismatch for " + mi + ".");
+            }
+            return mi;
+        }
+
+        private J.MethodInvocation withTemplate(J.MethodInvocation method, JavaTemplate template, Expression firstArg, List<Expression> matcherArgs) {
+            switch (matcherArgs.size()) {
+                case 0:
+                    return method.withTemplate(template, method.getCoordinates().replace(), firstArg);
+                case 1:
+                    return method.withTemplate(template, method.getCoordinates().replace(), firstArg, matcherArgs.get(0));
+                case 2 :
+                    return method.withTemplate(template, method.getCoordinates().replace(), firstArg, matcherArgs.get(0), matcherArgs.get(1));
+                case 3 :
+                    return method.withTemplate(template, method.getCoordinates().replace(), firstArg, matcherArgs.get(0), matcherArgs.get(1), matcherArgs.get(2));
+                default:
+                    throw new IllegalArgumentException("List of matcher arguments is too long for " + method + ".");
+            }
+        }
+
+        private List<Expression> stripMatcherInvocation(Expression e) {
+            if (e instanceof J.MethodInvocation) {
+                MethodMatcher matchesMatcher = new MethodMatcher("org.hamcrest.Matchers *(..)");
+                if (matchesMatcher.matches(e)) {
+                    return ((J.MethodInvocation) e).getArguments();
+                }
+            }
+            throw new IllegalArgumentException("Trying to strip an expression which is not a matcher invocation:\n" + e);
+        }
+
+        private String getTranslatedAssert(J.MethodInvocation methodInvocation) {
+            //to be replaced with a static map
+            switch (methodInvocation.getSimpleName()) {
+                case "equalTo":
+                    return "assertEquals";
+            }
+            throw new IllegalArgumentException("Translation of matcher " + methodInvocation.getSimpleName() + " not yet supported.");
+        }
+
+        private String getTemplateForTranslatedAssertion(String translatedAssertion) {
+            //to be replaced with a static map
+            switch (translatedAssertion) {
+                case "assertEquals":
+                    return "assertEquals(#{any(java.lang.Object)}, #{any(java.lang.Object)})";
```

Comment:
I think the recipe will fail currently with any 3/4 method arguments, as the template hardcodes the expected number of arguments to two; Might be best to add tests for various amounts of arguments and work from there to repeatedly append `, #{any(java.lang.Object)}` as needed, as also seen here.
https://github.com/openrewrite/rewrite-testing-frameworks/blob/f25ff0b0ed2b9e227b07972986b35f7a1cc958af/src/main/java/org/openrewrite/java/testing/cleanup/AssertTrueEqualsToAssertEquals.java#L74-L82